rir/src/compiler/translations/pir_2_rir/pir_2_rir.cpp

```diff
@@ -1243,12 +1290,12 @@ rir::Code* Pir2Rir::compileCode(Context& ctx, Code* code) {
             }
             }
 
-            if (instr->needsReferenceCount()) {
-                if (needsRefcount[instr] == AUses::Multiple)
-                    cb.add(BC::ensureNamed());
-                else if (needsRefcount[instr] == AUses::Destructive)
-                    cb.add(BC::setShared());
-            }
+            if (instr->minReferenceCount() < 2 && needsSetShared.count(instr))
+                cb.add(BC::setShared());
+            else if (instr->minReferenceCount() < 1 &&
+                     (refcountAnalysisOverflow ||
```

Comment:
the refcount analysis is only concerned with NAMED 0 vs. 1.  To decide 1 vs. 2 a simpler and more conservative approach is needed, because a precise approach would need to track bindings...

Reply:
Sorry to nitpick again, but can you add a comment for this? I still think the overflow behaviour is a bit surprising. (Just sneak it into your next commit or commit directly to master.)